projects/packages/forms/src/contact-form/class-contact-form-plugin.php

```diff
@@ -508,8 +500,26 @@ public static function block_attributes_to_shortcode_attributes( $atts, $type, $
 						}
 					}
 
-					$atts['options']     = implode( ',', $options );
-					$atts['optionsdata'] = \wp_json_encode( $options_data );
+					$atts['options']                          = implode( ',', $options );
+					$atts['optionsdata']                      = \wp_json_encode( $options_data );
+					$add_block_style_classes_to_field_wrapper = true;
+				}
+			}
+
+			/*
+			 * Add the `wp-block-jetpack-field-*` and `is-style-*` classes to the field wrapper div
+			 * for fields that are one of the new inner block types.
+			 * This ensures any updates to field block styles in theme.json or global styles are
+			 * correctly applied.
+			 */
+			if ( $add_block_style_classes_to_field_wrapper ) {
+				$atts['fieldwrapperclasses'] = 'wp-block-jetpack-field-' . $type;
+				if ( ! empty( $atts['class'] ) ) {
+					$block_style_classes          = self::get_block_style_classes( $atts['class'] );
+					$spacer                       = ! empty( $block_style_classes['block_style_classes'] ) ? ' ' : '';
+					$atts['fieldwrapperclasses'] .= $spacer . $block_style_classes['block_style_classes'] . $spacer . $block_style_classes['wrap_classes'];
```

Comment:
Is there a case that ` $block_style_classes['block_style_classes']` could be empty, while `$block_style_classes['wrap_classes']` contains a value?

As in that situation it looks like it'll be appended straight onto `$atts['fieldwrapperclasses']` without any spaces between them.

Reply:
Good question. No, since if there are no block style classes there will be no wrap classes.

See: https://github.com/Automattic/jetpack/blob/update/only-apply-block-classes-to-new-inner-blocks-field-wrappers/projects/packages/forms/src/contact-form/class-contact-form-plugin.php#L403

I should probably make it more explicit or clear.

Reply:
Ok, I debugged it a bit. It looks like both contain the same thing, the block variation style classname, but one has  `-wrap` appended to the end.

It makes me wonder why `get_block_style_classes` returns them as separate properties if they end up appended together.

Reply:
> It makes me wonder why get_block_style_classes returns them as separate properties if they end up appended together.

Oh, yeah I think I was doing something else at some stage and forgot to join them 😄 

If you want to join in on the confusion, take a look at [render_field](https://github.com/Automattic/jetpack/blob/update/only-apply-block-classes-to-new-inner-blocks-field-wrappers/projects/packages/forms/src/contact-form/class-contact-form-field.php#L1494) - it does some funky stuff. 

I'll update tomorrow.

Reply:
I merged the field classes and wrap in [b990885](https://github.com/Automattic/jetpack/pull/43268/commits/b990885f9ca20d2ce0d6f7a873576b466b64b5d2)